utils/docker-ce/files/dockerd.init

```diff
@@ -30,17 +87,21 @@ process_config() {
 
 	config_get data_root globals data_root "/opt/docker/"
 	config_get log_level globals log_level "warn"
+	config_get bip globals bip "172.17.0.1/16"
```

Comment:
We need a default because we add the bridge to the UCI right?
I'm just asking because we don't have a way of having of adding config options without a default yet.
we'd need something like `[ -n "$option" ] && json_add_string "option" "$option"` or something nicer so that we aren't maintaining defaults over the ones build into the daemon.

Reply:
I do not understand you exactly? Sorry
But the code can be described as follows:
If no config option **bip** is set in die uci config (/etc/config/dockerd) then use `172.17.0.1/16` as default bip value.

Reply:
I mean `daemon.json` has a lot of options and if we have to figure out the default for each we one we add, then also maintain its the same as the default built into the daemon. e.g making sure `{ "bip": "172.17.0.1/16" }` in the `daemon.json` has the same effect as leaving it out of the `daemon.json`. it'll quickly add up to a lot of maintenance, unless I'm mistaken.

But from what you said in the other comments, it sounds like we need to know the default or otherwise so we can set up `docker0`.
This is just a consideration for options other than these is what I'm thinking.

Reply:
> But from what you said in the other comments, it sounds like we need to know the default or otherwise so we can set up `docker0`.
> This is just a consideration for options other than these is what I'm thinking.

That's right. Also, it seems that if I set the option `bip`, the firewall chain `DOCKER-USER` is not created.
I agree with you about the other option. We should not maintain the default.